tests/cmd/auth/test_auth_config.py

```diff
@@ -0,0 +1,442 @@
+from datetime import datetime, timezone
+from typing import Tuple
+
+import pytest
+
+from ggshield.cmd.main import cli
+from ggshield.core.config import Config
+
+from .utils import prepare_config
+
+
+DEFAULT_INSTANCE_URL = "https://dashboard.gitguardian.com"
+
+EXPECTED_OUTPUT = """[https://dashboard.gitguardian.com]
+default_token_lifetime: None
+workspace: 1
+url: https://dashboard.gitguardian.com
+token: some token
+token_name: some token name
+expiry: 2022-05-04T17:00:00Z
+
+[https://some-gg-instance.com]
+default_token_lifetime: None
+workspace: 1
+url: https://some-gg-instance.com
+token: some token
+token_name: first token
+expiry: 2022-05-04T17:00:00Z
+
+[https://some-gg-instance.com]
+default_token_lifetime: None
+workspace: not set
+url: https://some-gg-instance.com
+token: not set
+token_name: not set
+expiry: not set
+"""
+
+
+@pytest.fixture(autouse=True)
+def tmp_config(monkeypatch, tmp_path):
+    monkeypatch.setattr(
+        "ggshield.core.config.get_auth_config_dir", lambda: str(tmp_path)
+    )
```

Comment:
:question: why do you need this? most (all?) tests are run with a temporary filesystem already

Reply:
otherwise it uses the local config so tests data is not truly isolated